qiskit/circuit/library/n_local/qaoa_ansatz.py

```diff
@@ -21,8 +21,72 @@
 from qiskit.circuit.quantumcircuit import QuantumCircuit
 from qiskit.circuit.quantumregister import QuantumRegister
 from qiskit.quantum_info import SparsePauliOp
+from qiskit.quantum_info.operators.base_operator import BaseOperator
+
+from .evolved_operator_ansatz import (
+    EvolvedOperatorAnsatz,
+    _is_pauli_identity,
+    evolved_operator_ansatz,
+)
+
+
+def qaoa_ansatz(
+    cost_operator: BaseOperator,
+    reps: int = 1,
+    initial_state: QuantumCircuit | None = None,
+    mixer_operator: BaseOperator | None = None,
+    insert_barriers: bool = False,
+    name: str = "QAOA",
+    flatten: bool = True,
+):
+    r"""A generalized QAOA quantum circuit with a support of custom initial states and mixers.
```

Comment:
might be good to an some more explanation or example here

Reply:
Another comment - please note that these functions do not appear in the table with the classes, as in #13310

Reply:
done in 076f467